src/lib.rs

```diff
@@ -1,19 +1,22 @@
 #![no_std]
 #![allow(non_camel_case_types)]
 
-#[cfg(not(feature = "device-selected"))]
+#[cfg(all(not(feature = "device-selected"), not(feature = "needs-subvariant")))]
 compile_error!(
-    "This crate requires one of the following device features enabled:
-        stm32f301
-        stm32f318
-        stm32f302
-        stm32f303
-        stm32f373
-        stm32f378
-        stm32f334
-        stm32f328
-        stm32f358
-        stm32f398"
+    "This crate requires you to specify your target device as a feature.
```

Comment:
I do understand your intention. This would remove the work to update this file, if a new feature gets added. 

However, I'd like to keep the list. Or at least update it to the current list of features avaliable. 
I think it is more helpful, when rustc directly tells you, which feature to use, without having to lookup the README on github / docs.rs or crates.io. 

Suggesting that there are more details in the README is a nice addition though :) 

Thinking about it we don't even include the README in `Cargo.toml` yet and I like to see a feature explanation as a doc comment on top of `lib.rs`. But this can be implemented in a follow up PR :)

Reply:
As much as I don't like to have information duplication (as it is a given that the information will diverge some day) I love the compiler telling me what to do to solve my problem.

@Sh3Rm4n 
> […] I like to see a feature explanation as a doc comment on top of `lib.rs`.

How will this be displayed in the docs?
And what is the benefit over just putting it into a compile error?

Reply:
nice. → https://github.com/stm32-rs/stm32f3xx-hal/pull/75/commits/ffa495a473a0769ee127ca8739ce30c6ccac9da6